caching.py

```
@@ -0,0 +1,613 @@
+        return None
+    
+    # Try to get from cache
+    cached_result = cache_manager.get(shared)
+    if cached_result is not None:
+        return cached_result
+    
+    return cache_manager, ttl or cache_manager.default_ttl
+
+
+@contextmanager
+def cache_context(shared: Dict[str, Any], ttl: Optional[float] = None):
```

Ruff (0.12.2)

Missing return type annotation for public function `cache_context`

(ANN201)